libraries/hatch-server-user-management/src/UserServiceClient.ts

```diff
@@ -0,0 +1,23 @@
+export class UserInfo {
+  public userId?: string;
+  public username?: string;
+  public accessToken?: string;
+  public isAuthenticated: boolean = false;
+}
+
+export class AuthTokens {
+  public accessToken?: string;
+  public refreshToken?: string;
+}
+
+export default interface UserServiceClient {
+  authenticate(username: string, password: string): Promise<AuthTokens>;
+  signUpUser(username: string, password: string, userAttributes?: {[key: string]: any}): Promise<void>;
+  resendSignUp(username: string): Promise<void>;
```

Comment:
I think we can clean these names up a bit and make them consistent throughout the codebase -- elsewhere we're using "reset password", etc. I'd suggest the following names:

- authenticate
- startUserRegistration
- resendUserRegistrationCode
- confirmUserRegistration
- startPasswordReset
- confirmPasswordReset
- getUserInfo (instead of verifyToken)
- refreshAuthentication
- signOutUser

I think we're also missing these?:
- getUserAttributes
- setUserAttributes

Reply:
Updated